{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    ".libPaths('/cellar/users/e5silva/anaconda3/envs/r3_3/lib/R/library/') \n",
    "library('rentrez')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'pubmed'</li>\n",
       "\t<li>'protein'</li>\n",
       "\t<li>'nuccore'</li>\n",
       "\t<li>'ipg'</li>\n",
       "\t<li>'nucleotide'</li>\n",
       "\t<li>'nucgss'</li>\n",
       "\t<li>'nucest'</li>\n",
       "\t<li>'structure'</li>\n",
       "\t<li>'sparcle'</li>\n",
       "\t<li>'genome'</li>\n",
       "\t<li>'annotinfo'</li>\n",
       "\t<li>'assembly'</li>\n",
       "\t<li>'bioproject'</li>\n",
       "\t<li>'biosample'</li>\n",
       "\t<li>'blastdbinfo'</li>\n",
       "\t<li>'books'</li>\n",
       "\t<li>'cdd'</li>\n",
       "\t<li>'clinvar'</li>\n",
       "\t<li>'clone'</li>\n",
       "\t<li>'gap'</li>\n",
       "\t<li>'gapplus'</li>\n",
       "\t<li>'grasp'</li>\n",
       "\t<li>'dbvar'</li>\n",
       "\t<li>'gene'</li>\n",
       "\t<li>'gds'</li>\n",
       "\t<li>'geoprofiles'</li>\n",
       "\t<li>'homologene'</li>\n",
       "\t<li>'medgen'</li>\n",
       "\t<li>'mesh'</li>\n",
       "\t<li>'ncbisearch'</li>\n",
       "\t<li>'nlmcatalog'</li>\n",
       "\t<li>'omim'</li>\n",
       "\t<li>'orgtrack'</li>\n",
       "\t<li>'pmc'</li>\n",
       "\t<li>'popset'</li>\n",
       "\t<li>'probe'</li>\n",
       "\t<li>'proteinclusters'</li>\n",
       "\t<li>'pcassay'</li>\n",
       "\t<li>'biosystems'</li>\n",
       "\t<li>'pccompound'</li>\n",
       "\t<li>'pcsubstance'</li>\n",
       "\t<li>'pubmedhealth'</li>\n",
       "\t<li>'seqannot'</li>\n",
       "\t<li>'snp'</li>\n",
       "\t<li>'sra'</li>\n",
       "\t<li>'taxonomy'</li>\n",
       "\t<li>'biocollections'</li>\n",
       "\t<li>'unigene'</li>\n",
       "\t<li>'gencoll'</li>\n",
       "\t<li>'gtr'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'pubmed'\n",
       "\\item 'protein'\n",
       "\\item 'nuccore'\n",
       "\\item 'ipg'\n",
       "\\item 'nucleotide'\n",
       "\\item 'nucgss'\n",
       "\\item 'nucest'\n",
       "\\item 'structure'\n",
       "\\item 'sparcle'\n",
       "\\item 'genome'\n",
       "\\item 'annotinfo'\n",
       "\\item 'assembly'\n",
       "\\item 'bioproject'\n",
       "\\item 'biosample'\n",
       "\\item 'blastdbinfo'\n",
       "\\item 'books'\n",
       "\\item 'cdd'\n",
       "\\item 'clinvar'\n",
       "\\item 'clone'\n",
       "\\item 'gap'\n",
       "\\item 'gapplus'\n",
       "\\item 'grasp'\n",
       "\\item 'dbvar'\n",
       "\\item 'gene'\n",
       "\\item 'gds'\n",
       "\\item 'geoprofiles'\n",
       "\\item 'homologene'\n",
       "\\item 'medgen'\n",
       "\\item 'mesh'\n",
       "\\item 'ncbisearch'\n",
       "\\item 'nlmcatalog'\n",
       "\\item 'omim'\n",
       "\\item 'orgtrack'\n",
       "\\item 'pmc'\n",
       "\\item 'popset'\n",
       "\\item 'probe'\n",
       "\\item 'proteinclusters'\n",
       "\\item 'pcassay'\n",
       "\\item 'biosystems'\n",
       "\\item 'pccompound'\n",
       "\\item 'pcsubstance'\n",
       "\\item 'pubmedhealth'\n",
       "\\item 'seqannot'\n",
       "\\item 'snp'\n",
       "\\item 'sra'\n",
       "\\item 'taxonomy'\n",
       "\\item 'biocollections'\n",
       "\\item 'unigene'\n",
       "\\item 'gencoll'\n",
       "\\item 'gtr'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'pubmed'\n",
       "2. 'protein'\n",
       "3. 'nuccore'\n",
       "4. 'ipg'\n",
       "5. 'nucleotide'\n",
       "6. 'nucgss'\n",
       "7. 'nucest'\n",
       "8. 'structure'\n",
       "9. 'sparcle'\n",
       "10. 'genome'\n",
       "11. 'annotinfo'\n",
       "12. 'assembly'\n",
       "13. 'bioproject'\n",
       "14. 'biosample'\n",
       "15. 'blastdbinfo'\n",
       "16. 'books'\n",
       "17. 'cdd'\n",
       "18. 'clinvar'\n",
       "19. 'clone'\n",
       "20. 'gap'\n",
       "21. 'gapplus'\n",
       "22. 'grasp'\n",
       "23. 'dbvar'\n",
       "24. 'gene'\n",
       "25. 'gds'\n",
       "26. 'geoprofiles'\n",
       "27. 'homologene'\n",
       "28. 'medgen'\n",
       "29. 'mesh'\n",
       "30. 'ncbisearch'\n",
       "31. 'nlmcatalog'\n",
       "32. 'omim'\n",
       "33. 'orgtrack'\n",
       "34. 'pmc'\n",
       "35. 'popset'\n",
       "36. 'probe'\n",
       "37. 'proteinclusters'\n",
       "38. 'pcassay'\n",
       "39. 'biosystems'\n",
       "40. 'pccompound'\n",
       "41. 'pcsubstance'\n",
       "42. 'pubmedhealth'\n",
       "43. 'seqannot'\n",
       "44. 'snp'\n",
       "45. 'sra'\n",
       "46. 'taxonomy'\n",
       "47. 'biocollections'\n",
       "48. 'unigene'\n",
       "49. 'gencoll'\n",
       "50. 'gtr'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"pubmed\"          \"protein\"         \"nuccore\"         \"ipg\"            \n",
       " [5] \"nucleotide\"      \"nucgss\"          \"nucest\"          \"structure\"      \n",
       " [9] \"sparcle\"         \"genome\"          \"annotinfo\"       \"assembly\"       \n",
       "[13] \"bioproject\"      \"biosample\"       \"blastdbinfo\"     \"books\"          \n",
       "[17] \"cdd\"             \"clinvar\"         \"clone\"           \"gap\"            \n",
       "[21] \"gapplus\"         \"grasp\"           \"dbvar\"           \"gene\"           \n",
       "[25] \"gds\"             \"geoprofiles\"     \"homologene\"      \"medgen\"         \n",
       "[29] \"mesh\"            \"ncbisearch\"      \"nlmcatalog\"      \"omim\"           \n",
       "[33] \"orgtrack\"        \"pmc\"             \"popset\"          \"probe\"          \n",
       "[37] \"proteinclusters\" \"pcassay\"         \"biosystems\"      \"pccompound\"     \n",
       "[41] \"pcsubstance\"     \"pubmedhealth\"    \"seqannot\"        \"snp\"            \n",
       "[45] \"sra\"             \"taxonomy\"        \"biocollections\"  \"unigene\"        \n",
       "[49] \"gencoll\"         \"gtr\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "entrez_dbs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$uid</dt>\n",
       "\t\t<dd>'100124402'</dd>\n",
       "\t<dt>$name</dt>\n",
       "\t\t<dd>'LOC100124402'</dd>\n",
       "\t<dt>$description</dt>\n",
       "\t\t<dd>'ubiquitin-fold modifier conjugating enzyme 1 pseudogene'</dd>\n",
       "\t<dt>$status</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$currentid</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$chromosome</dt>\n",
       "\t\t<dd>'7'</dd>\n",
       "\t<dt>$geneticsource</dt>\n",
       "\t\t<dd>'genomic'</dd>\n",
       "\t<dt>$maplocation</dt>\n",
       "\t\t<dd>'7q21.11'</dd>\n",
       "\t<dt>$otheraliases</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$otherdesignations</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$nomenclaturesymbol</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$nomenclaturename</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$nomenclaturestatus</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$mim</dt>\n",
       "\t\t<dd><ol>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$genomicinfo</dt>\n",
       "\t\t<dd><table>\n",
       "<thead><tr><th scope=col>chrloc</th><th scope=col>chraccver</th><th scope=col>chrstart</th><th scope=col>chrstop</th><th scope=col>exoncount</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>7           </td><td>NC_000007.14</td><td>78486515    </td><td>78487399    </td><td>            </td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "\t<dt>$geneweight</dt>\n",
       "\t\t<dd>23</dd>\n",
       "\t<dt>$summary</dt>\n",
       "\t\t<dd>''</dd>\n",
       "\t<dt>$chrsort</dt>\n",
       "\t\t<dd>'07'</dd>\n",
       "\t<dt>$chrstart</dt>\n",
       "\t\t<dd>78486515</dd>\n",
       "\t<dt>$organism</dt>\n",
       "\t\t<dd><dl>\n",
       "\t<dt>$scientificname</dt>\n",
       "\t\t<dd>'Homo sapiens'</dd>\n",
       "\t<dt>$commonname</dt>\n",
       "\t\t<dd>'human'</dd>\n",
       "\t<dt>$taxid</dt>\n",
       "\t\t<dd>9606</dd>\n",
       "</dl>\n",
       "</dd>\n",
       "\t<dt>$locationhist</dt>\n",
       "\t\t<dd><table>\n",
       "<thead><tr><th scope=col>annotationrelease</th><th scope=col>assemblyaccver</th><th scope=col>chraccver</th><th scope=col>chrstart</th><th scope=col>chrstop</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>109             </td><td>GCF_000001405.38</td><td>NC_000007.14    </td><td>78486515        </td><td>78487399        </td></tr>\n",
       "\t<tr><td>108             </td><td>GCF_000001405.33</td><td>NC_000007.14    </td><td>78486515        </td><td>78487399        </td></tr>\n",
       "\t<tr><td>108             </td><td>GCF_000306695.2 </td><td>NC_018918.2     </td><td>78046284        </td><td>78047167        </td></tr>\n",
       "\t<tr><td>107             </td><td>GCF_000001405.28</td><td>NC_000007.14    </td><td>78486515        </td><td>78487399        </td></tr>\n",
       "\t<tr><td>107             </td><td>GCF_000306695.2 </td><td>NC_018918.2     </td><td>78046284        </td><td>78047167        </td></tr>\n",
       "\t<tr><td>106             </td><td>GCF_000001405.26</td><td>NC_000007.14    </td><td>78486515        </td><td>78487399        </td></tr>\n",
       "\t<tr><td>106             </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>106             </td><td>GCF_000306695.2 </td><td>NC_018918.2     </td><td>78046284        </td><td>78047167        </td></tr>\n",
       "\t<tr><td>105             </td><td>GCF_000001405.25</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>105             </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>105             </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>105             </td><td>GCF_000306695.2 </td><td>NC_018918.2     </td><td>78046284        </td><td>78047167        </td></tr>\n",
       "\t<tr><td>104             </td><td>GCF_000001405.22</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>104             </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>104             </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>104             </td><td>GCF_000306695.1 </td><td>NC_018918.1     </td><td>81424078        </td><td>81424962        </td></tr>\n",
       "\t<tr><td>103             </td><td>GCF_000001405.21</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>103             </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>103             </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>37.3            </td><td>GCF_000001405.17</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>37.3            </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>37.3            </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>37.2            </td><td>GCF_000001405.14</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>37.2            </td><td>GCF_000002115.2 </td><td>AC_000050.1     </td><td>72816115        </td><td>72816997        </td></tr>\n",
       "\t<tr><td>37.2            </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>37.2            </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>37.1            </td><td>GCF_000001405.13</td><td>NC_000007.13    </td><td>78115832        </td><td>78116716        </td></tr>\n",
       "\t<tr><td>37.1            </td><td>GCF_000002115.2 </td><td>AC_000050.1     </td><td>72816115        </td><td>72816997        </td></tr>\n",
       "\t<tr><td>37.1            </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448418        </td></tr>\n",
       "\t<tr><td>37.1            </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718840        </td></tr>\n",
       "\t<tr><td>36.3            </td><td>GCF_000001405.12</td><td>NC_000007.12    </td><td>77953768        </td><td>77954647        </td></tr>\n",
       "\t<tr><td>36.3            </td><td>GCF_000002115.2 </td><td>AC_000050.1     </td><td>72816115        </td><td>72816992        </td></tr>\n",
       "\t<tr><td>36.3            </td><td>GCF_000002135.2 </td><td>AC_000068.1     </td><td>77447536        </td><td>77448413        </td></tr>\n",
       "\t<tr><td>36.3            </td><td>GCF_000002125.1 </td><td>AC_000139.1     </td><td>72717958        </td><td>72718835        </td></tr>\n",
       "</tbody>\n",
       "</table>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$uid] '100124402'\n",
       "\\item[\\$name] 'LOC100124402'\n",
       "\\item[\\$description] 'ubiquitin-fold modifier conjugating enzyme 1 pseudogene'\n",
       "\\item[\\$status] ''\n",
       "\\item[\\$currentid] ''\n",
       "\\item[\\$chromosome] '7'\n",
       "\\item[\\$geneticsource] 'genomic'\n",
       "\\item[\\$maplocation] '7q21.11'\n",
       "\\item[\\$otheraliases] ''\n",
       "\\item[\\$otherdesignations] ''\n",
       "\\item[\\$nomenclaturesymbol] ''\n",
       "\\item[\\$nomenclaturename] ''\n",
       "\\item[\\$nomenclaturestatus] ''\n",
       "\\item[\\$mim] \\begin{enumerate}\n",
       "\\end{enumerate}\n",
       "\n",
       "\\item[\\$genomicinfo] \\begin{tabular}{r|lllll}\n",
       " chrloc & chraccver & chrstart & chrstop & exoncount\\\\\n",
       "\\hline\n",
       "\t 7              & NC\\_000007.14 & 78486515       & 78487399       &               \\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\item[\\$geneweight] 23\n",
       "\\item[\\$summary] ''\n",
       "\\item[\\$chrsort] '07'\n",
       "\\item[\\$chrstart] 78486515\n",
       "\\item[\\$organism] \\begin{description}\n",
       "\\item[\\$scientificname] 'Homo sapiens'\n",
       "\\item[\\$commonname] 'human'\n",
       "\\item[\\$taxid] 9606\n",
       "\\end{description}\n",
       "\n",
       "\\item[\\$locationhist] \\begin{tabular}{r|lllll}\n",
       " annotationrelease & assemblyaccver & chraccver & chrstart & chrstop\\\\\n",
       "\\hline\n",
       "\t 109                & GCF\\_000001405.38 & NC\\_000007.14     & 78486515           & 78487399          \\\\\n",
       "\t 108                & GCF\\_000001405.33 & NC\\_000007.14     & 78486515           & 78487399          \\\\\n",
       "\t 108                & GCF\\_000306695.2  & NC\\_018918.2      & 78046284           & 78047167          \\\\\n",
       "\t 107                & GCF\\_000001405.28 & NC\\_000007.14     & 78486515           & 78487399          \\\\\n",
       "\t 107                & GCF\\_000306695.2  & NC\\_018918.2      & 78046284           & 78047167          \\\\\n",
       "\t 106                & GCF\\_000001405.26 & NC\\_000007.14     & 78486515           & 78487399          \\\\\n",
       "\t 106                & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 106                & GCF\\_000306695.2  & NC\\_018918.2      & 78046284           & 78047167          \\\\\n",
       "\t 105                & GCF\\_000001405.25 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 105                & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 105                & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 105                & GCF\\_000306695.2  & NC\\_018918.2      & 78046284           & 78047167          \\\\\n",
       "\t 104                & GCF\\_000001405.22 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 104                & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 104                & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 104                & GCF\\_000306695.1  & NC\\_018918.1      & 81424078           & 81424962          \\\\\n",
       "\t 103                & GCF\\_000001405.21 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 103                & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 103                & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 37.3               & GCF\\_000001405.17 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 37.3               & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 37.3               & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 37.2               & GCF\\_000001405.14 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 37.2               & GCF\\_000002115.2  & AC\\_000050.1      & 72816115           & 72816997          \\\\\n",
       "\t 37.2               & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 37.2               & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 37.1               & GCF\\_000001405.13 & NC\\_000007.13     & 78115832           & 78116716          \\\\\n",
       "\t 37.1               & GCF\\_000002115.2  & AC\\_000050.1      & 72816115           & 72816997          \\\\\n",
       "\t 37.1               & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448418          \\\\\n",
       "\t 37.1               & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718840          \\\\\n",
       "\t 36.3               & GCF\\_000001405.12 & NC\\_000007.12     & 77953768           & 77954647          \\\\\n",
       "\t 36.3               & GCF\\_000002115.2  & AC\\_000050.1      & 72816115           & 72816992          \\\\\n",
       "\t 36.3               & GCF\\_000002135.2  & AC\\_000068.1      & 77447536           & 77448413          \\\\\n",
       "\t 36.3               & GCF\\_000002125.1  & AC\\_000139.1      & 72717958           & 72718835          \\\\\n",
       "\\end{tabular}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$uid\n",
       ":   '100124402'\n",
       "$name\n",
       ":   'LOC100124402'\n",
       "$description\n",
       ":   'ubiquitin-fold modifier conjugating enzyme 1 pseudogene'\n",
       "$status\n",
       ":   ''\n",
       "$currentid\n",
       ":   ''\n",
       "$chromosome\n",
       ":   '7'\n",
       "$geneticsource\n",
       ":   'genomic'\n",
       "$maplocation\n",
       ":   '7q21.11'\n",
       "$otheraliases\n",
       ":   ''\n",
       "$otherdesignations\n",
       ":   ''\n",
       "$nomenclaturesymbol\n",
       ":   ''\n",
       "$nomenclaturename\n",
       ":   ''\n",
       "$nomenclaturestatus\n",
       ":   ''\n",
       "$mim\n",
       ":   \n",
       "\n",
       "\n",
       "$genomicinfo\n",
       ":   \n",
       "chrloc | chraccver | chrstart | chrstop | exoncount | \n",
       "|---|\n",
       "| 7            | NC_000007.14 | 78486515     | 78487399     |              | \n",
       "\n",
       "\n",
       "\n",
       "$geneweight\n",
       ":   23\n",
       "$summary\n",
       ":   ''\n",
       "$chrsort\n",
       ":   '07'\n",
       "$chrstart\n",
       ":   78486515\n",
       "$organism\n",
       ":   $scientificname\n",
       ":   'Homo sapiens'\n",
       "$commonname\n",
       ":   'human'\n",
       "$taxid\n",
       ":   9606\n",
       "\n",
       "\n",
       "\n",
       "$locationhist\n",
       ":   \n",
       "annotationrelease | assemblyaccver | chraccver | chrstart | chrstop | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 109              | GCF_000001405.38 | NC_000007.14     | 78486515         | 78487399         | \n",
       "| 108              | GCF_000001405.33 | NC_000007.14     | 78486515         | 78487399         | \n",
       "| 108              | GCF_000306695.2  | NC_018918.2      | 78046284         | 78047167         | \n",
       "| 107              | GCF_000001405.28 | NC_000007.14     | 78486515         | 78487399         | \n",
       "| 107              | GCF_000306695.2  | NC_018918.2      | 78046284         | 78047167         | \n",
       "| 106              | GCF_000001405.26 | NC_000007.14     | 78486515         | 78487399         | \n",
       "| 106              | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 106              | GCF_000306695.2  | NC_018918.2      | 78046284         | 78047167         | \n",
       "| 105              | GCF_000001405.25 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 105              | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 105              | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 105              | GCF_000306695.2  | NC_018918.2      | 78046284         | 78047167         | \n",
       "| 104              | GCF_000001405.22 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 104              | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 104              | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 104              | GCF_000306695.1  | NC_018918.1      | 81424078         | 81424962         | \n",
       "| 103              | GCF_000001405.21 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 103              | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 103              | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 37.3             | GCF_000001405.17 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 37.3             | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 37.3             | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 37.2             | GCF_000001405.14 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 37.2             | GCF_000002115.2  | AC_000050.1      | 72816115         | 72816997         | \n",
       "| 37.2             | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 37.2             | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 37.1             | GCF_000001405.13 | NC_000007.13     | 78115832         | 78116716         | \n",
       "| 37.1             | GCF_000002115.2  | AC_000050.1      | 72816115         | 72816997         | \n",
       "| 37.1             | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448418         | \n",
       "| 37.1             | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718840         | \n",
       "| 36.3             | GCF_000001405.12 | NC_000007.12     | 77953768         | 77954647         | \n",
       "| 36.3             | GCF_000002115.2  | AC_000050.1      | 72816115         | 72816992         | \n",
       "| 36.3             | GCF_000002135.2  | AC_000068.1      | 77447536         | 77448413         | \n",
       "| 36.3             | GCF_000002125.1  | AC_000139.1      | 72717958         | 72718835         | \n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "esummary result with 20 items:\n",
       " [1] uid                name               description        status            \n",
       " [5] currentid          chromosome         geneticsource      maplocation       \n",
       " [9] otheraliases       otherdesignations  nomenclaturesymbol nomenclaturename  \n",
       "[13] nomenclaturestatus mim                genomicinfo        geneweight        \n",
       "[17] summary            chrsort            chrstart           organism          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# testing getting out one record\n",
    "r_search <- entrez_summary(db='gene', id=100124402)\n",
    "r_search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2650"
      ],
      "text/latex": [
       "2650"
      ],
      "text/markdown": [
       "2650"
      ],
      "text/plain": [
       "[1] 2650"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>100124402</li>\n",
       "\t<li>100126583</li>\n",
       "\t<li>100127889</li>\n",
       "\t<li>100127910</li>\n",
       "\t<li>100127955</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 100124402\n",
       "\\item 100126583\n",
       "\\item 100127889\n",
       "\\item 100127910\n",
       "\\item 100127955\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 100124402\n",
       "2. 100126583\n",
       "3. 100127889\n",
       "4. 100127910\n",
       "5. 100127955\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 100124402 100126583 100127889 100127910 100127955"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# import list\n",
    "not_there <- read.table('not_there.txt', header=FALSE)\n",
    "dim(not_there)[1]\n",
    "not_there[1:5,1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# look up all names \n",
    "now_there <- data.frame(Entrez_ID=character(), Symbol=character(), stringsAsFactors=FALSE)\n",
    "\n",
    "for (i in 1:dim(not_there)[1]){\n",
    "    r_search <- entrez_summary(db='gene', id=not_there[i,1])\n",
    "    r_search <- r_search$name\n",
    "    \n",
    "    now_there[i,1] <- not_there[i,1]\n",
    "    now_there[i,2] <- r_search\n",
    "    r_search <- ''\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>Entrez_ID</th><th scope=col>Symbol</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>100124402   </td><td>LOC100124402</td></tr>\n",
       "\t<tr><td>100126583   </td><td>LOC100126583</td></tr>\n",
       "\t<tr><td>100127889   </td><td>C10orf131   </td></tr>\n",
       "\t<tr><td>100127910   </td><td>LOC100127910</td></tr>\n",
       "\t<tr><td>100127955   </td><td>LOC100127955</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|ll}\n",
       " Entrez\\_ID & Symbol\\\\\n",
       "\\hline\n",
       "\t 100124402    & LOC100124402\\\\\n",
       "\t 100126583    & LOC100126583\\\\\n",
       "\t 100127889    & C10orf131   \\\\\n",
       "\t 100127910    & LOC100127910\\\\\n",
       "\t 100127955    & LOC100127955\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "Entrez_ID | Symbol | \n",
       "|---|---|---|---|---|\n",
       "| 100124402    | LOC100124402 | \n",
       "| 100126583    | LOC100126583 | \n",
       "| 100127889    | C10orf131    | \n",
       "| 100127910    | LOC100127910 | \n",
       "| 100127955    | LOC100127955 | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  Entrez_ID Symbol      \n",
       "1 100124402 LOC100124402\n",
       "2 100126583 LOC100126583\n",
       "3 100127889 C10orf131   \n",
       "4 100127910 LOC100127910\n",
       "5 100127955 LOC100127955"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "now_there[1:5,1:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(now_there, file = \"now_there.csv\", row.names=FALSE, col.names=FALSE, na=\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
